### PR TITLE
Booking engine configuration: payment card input deprecation

### PR DESCRIPTION
**Pull request by JakubSaksa**
#### Changelog notes 

Updated documentation because of changes related to this: https://mews.myjetbrains.com/youtrack/issue/RND-54542.

* Added new field to `configuration/get` response => `Configuration.PaymentCardRequirement`

#### Check during review

- [ ] JSON example extended.
  - [ ] New properties are added to the correct place in the JSON.
- [ ] New properties in the table are added to the correct place.
- [ ] Correct formatting:
  - [ ] Spacing is consistent between titles, sections, tables, ...
  - [ ] Correct JSON format - indentation.

### CODE REVIEW

File: distributor-api-v1/deprecations/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Deprecations
+
```
Comment (koprivajakub) [inline distributor-api-v1/deprecations/README.md:2]: Nice. Thx.

I would just add a paragraph:
```
This page contains a list of past and future deprecations in the API. Deprecation information are present in individual pages. `Deprecation notice` is a date from which we do support both versions (previous/deprecated and next/living standard). `Removal date` is a date where we aim to remove the previous/deprecated variant of the API.
```

cc @vitasamek maybe you can come up with some better phrasing 👍

File: distributor-api-v1/deprecations/payment-card-input.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
+# PaymentCardInput deprecation
+[PaymentCardInput](#paymentcardinput) field of [configuration](../operations.md#configuration) object is deprecated and will be removed from the response of [configuration/get](../operations.md#getconfigurationinfo) on 31.7.2021. It is replaced by [PaymentCardRequirement](#paymentcardrequirement). The new field contains two new options for requiring a payment card.
+
+## Response diff
+
+```diff
```
Comment (koprivajakub) [inline distributor-api-v1/deprecations/payment-card-input.md:6]: kudos for the `diff` syntax 👍 this format is exactly what I was looking for

File: distributor-api-v1/deprecations/payment-card-input.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    "StartDateOffset": null,
+    "Theme": null,
+    "VoucherCode": ""
+}
+```
+
+### PaymentCardInput
+* `NotRequested` - Payment card info is not requested.
+* `Requested` - Payment card info is requested, but not validated.
+* `Required` - Payment card info is requested and validated.
```
Comment (koprivajakub) [inline distributor-api-v1/deprecations/payment-card-input.md:148]: Not sure whether to mention the old version here. It is valuable to see what the options was, but I would somehow mark it as "invalid" maybe a ~~stiketrough~~ might do the job?

@vitasamek what do you think?

**Review by koprivajakub — APPROVED** [review]

File: SUMMARY.md [hunk 4]
```diff
@@ -16,6 +16,8 @@
   * [Images](distributor-api-v1/images.md)
   * [Payment Gateway Data](distributor-api-v1/payment-gateway-data.md)
   * [Environments](distributor-api-v1/environments.md)
+  * [Deprecations](distributor-api-v1/deprecations/README.md)
```
Comment (vitasamek) [inline SUMMARY.md:19]: Not sure if this is a good place, It's really hidden. I am afraid that people will skip it. I would reference it maybe on the top of the change log page.

File: distributor-api-v1/deprecations/README.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+This page contains a list of past and future deprecations in the API. Deprecation information are present in individual pages. `Deprecation notice` is a date from which we do support both versions (previous/deprecated and next/living standard). `Removal date` is a date where we aim to remove the previous/deprecated variant of the API.
+
+| Deprecation | Deprecation notice | Removal date |
+|:------------|:-------------------|:-------------|
+| [PaymentCardInput](payment-card-input.md) | 18.4.2021 | 31.7.2021 |
```
Comment (vitasamek) [inline distributor-api-v1/deprecations/README.md:7]: In the connector, we have a similar table where we have another column with a replacement. Then you don't have to write the long paragraph you have below the table.
image

File: distributor-api-v1/deprecations/payment-card-input.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
+# PaymentCardInput deprecation
+[PaymentCardInput](#paymentcardinput) field of [configuration](../operations.md#configuration) object is deprecated and will be removed from the response of [configuration/get](../operations.md#getconfigurationinfo) on 31.7.2021. It is replaced by [PaymentCardRequirement](#paymentcardrequirement). The new field contains two new options for requiring a payment card.
+
+## Response diff
+
+```diff
+{
```
Comment (vitasamek) [inline distributor-api-v1/deprecations/payment-card-input.md:7]: Is it worth having it? It's really long. If the change in the table refers to the object, the user can see the diff. Depends on you.

File: distributor-api-v1/deprecations/payment-card-input.md [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+### ~~PaymentCardInput~~
+* `NotRequested` - Payment card info is not requested.
+* `Requested` - Payment card info is requested, but not validated.
+* `Required` - Payment card info is requested and validated.
+
+### PaymentCardRequirement
+* `NotRequired` - Payment card info is never required. 
+* `AlwaysRequired` - Payment card info is always required and validated.
+* `NotRequiredForFullyPaidBookings` - Payment card info is not required for fully paid bookings. Otherwise required.
+* `NotRequiredForFullyOrPartiallyPaidBookings` - Payment card info is not required for fully or partially paid bookings. Otherwise required.
```
Comment (vitasamek) [inline distributor-api-v1/deprecations/payment-card-input.md:154]: This is a duplicity. You have the same text 2 times. If we will discover type, will you remember that you have to change here and in operations.md file?

File: distributor-api-v1/operations.md [hunk 8]
```diff
@@ -208,15 +208,15 @@ Preferred initial call used to obtain all static data about distributor configur
...
 | `ServiceId` | string | required | Unique identifier of the service to which the configuration is bound to. |
 
-#### PaymentCardInput
-
-* `NotRequested` - Payment card info is not requested.
-* `Requested` - Payment card info is requested, but not validated.
-* `Required` - Payment card info is requested and validated. 
+#### PaymentCardRequirement
+* `NotRequired` - Payment card info is never required. 
+* `AlwaysRequired` - Payment card info is always requested and validated.
```
Comment (vitasamek) [inline distributor-api-v1/operations.md:217]: why are we mixing together the required and requested word? :)
we should use only one to make it clear.

**Review by vitasamek — APPROVED** [review]

**Review by koprivajakub — APPROVED** [review]

**Pull request merged**